### DISCUSSION TITLE
Селект подмена значений

### DISCUSSION THREAD

**Discussion by Qb1cc** [totumonline/totum-ru-issues-and-discussions#83] [Как это сделать?] [answered]
Как сделать селект по артикулу, но с подменой на название при добавлении в список .

![image](https://user-images.githubusercontent.com/84081305/155281966-fb2a6195-50d5-426e-b827-da91f5c5f412.png)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Самый распространенный вариант склеить несколько полей из таблицы источника в отображение селекта — [вот видео с таймкодом —>](https://youtu.be/qmVTZ19rW1o?t=305)

Также можно в строке добавления использовать одно поле отображения, а в таблице другое, на основе [индивидуальных селектов для каждой строки —>](https://docs.totum.online/different-composition-of-select-depending-on-other-fields)

Строка добавления отличается от уже добавленный тем, что у нее `id = ""`. Те получается код селекта:

```
=: if(condition: #id = ""; then: $new_line; else: $exist_line)

new_line: selectRowListForSelect(table: 'table'; field: 'number')

exist_line: selectRowListForSelect(table: 'table'; field: 'title')

// включение Индивидуальный расчет селекта для каждой строки в этом случае обязательно

```

**Reply by Qb1cc** [reply]
Либо я не понял чего-то , либо это не совсем то что нужно. 
Есть самая простая таблица
![image](https://user-images.githubusercontent.com/84081305/155348827-3185da80-d912-49c4-91d7-42e3031aec59.png)
Я из цикла делаю селект где field: артикул ( с редактированием ).
![image](https://user-images.githubusercontent.com/84081305/155349264-dbdb568d-8118-459c-ab96-6e5c8f398937.png)
Но в итоге должен получить 
![image](https://user-images.githubusercontent.com/84081305/155349425-18189824-c5ee-4766-a827-561bbbc14cee.png)

А всё что я делаю, начинает работать полностью с колонкой наименование и в итоге получается такое
![image](https://user-images.githubusercontent.com/84081305/155349647-493bd813-feb8-4135-aac9-711c84cb3a76.png)

Очень важно сохранить возможность для менеджера выбирать товар по колонке артикул (без всяких наименований), но в итоге печатать только наименование.

**Reply by ttm-alexeypolunin** [reply]
Можно сделать разный селект для `строки добавления` и для `добавленной строки`:

<img width="1156" alt="Снимок экрана 2022-02-23 в 19 02 20" src="https://user-images.githubusercontent.com/55755565/155358021-913d5251-732c-4ff7-b951-831bcacaeabe.png">

Результат:

![Снимок экрана 2022-02-23 в 18 56 38](https://user-images.githubusercontent.com/55755565/155358123-fad5d5de-08a1-4e7f-a4d1-436914d86e86.png)

Сделать разное Отображение селекта для закрытой и открытой ячейки нельзя. Можно поверх имеющегося отображения использовать форматирование поля функции `setFormat` параметр `text`: [урок курса про text —>](https://docs.totum.online/what-are-text-and-coment-in-field-formatting)

Возможно еще понадобится сказать `textasvalue: true`

**Reply by Qb1cc** [reply]
В таком случае, перестаёт работать оприходование товара((
https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/75  

Решил просто доп поле сделать и в него пихать наименование из селекта, главное теперь с печатью разобраться. Как таблицу не целиком печатать а по колонкам нужным.
![image](https://user-images.githubusercontent.com/84081305/155374365-5eb7b1fb-2098-4811-b3a3-1c1aa0219c27.png)

**Reply by erpic-ru** [reply]
Самый очевидный способ - это добавить печатную форму (отдельную кнопку печати с кодом linktoprint , а не печать страницы) и сделать 2 поля в табличной части: артикул и подбираемое кодом нередактируемое через web наименование (select по артикулу).

При печати передавать все, кроме артикулов
[Про печать](https://docs.totum.online/print-and-csv)

**Reply by ttm-alexeypolunin** [reply]
Ну самый топорный способ (он же самый простой) — это ткнуть в глаз рядом с названием таблицы — там отключить ненужные поля — сохранить этот вид там в глазе как «на печать» — сделать его доступным для всех — переключатся в вид для печати и нажимать на печать.

Это чесн самый нищий вариант (это даж не доширак, а роллтон), но он у нас реально в паре проектов существует.

**Reply by Qb1cc** [reply]
Это совсем уж костыли)) Проще с печатью разобраться и кнопкой с шаблоном печатать. (html+css знаю) вчера научился передавать всё в печать, кроме табличной части ( но я нашёл вроде в уроках куда копать, думаю сегодня докопаюсь )

**Reply by Qb1cc** [reply]
![image](https://user-images.githubusercontent.com/84081305/155487290-57b15e15-b37c-414b-a860-ea99b91eff1b.png)
Помогите плиз, не хочет у меня список печататься и всё тут. 

Нужно вывести в печать отмеченные колонки. Создал два шаблона, из футера всё печатается таким кодом в кнопке, но вот список из таблицы в цикле, не хочет в печать уходить
`= : linkToPrint(template: "main_h"; data: $data)
 data: rowCreate(field: "data" = #h_h_data_zakaza; field: "pokupatel" = #smsto; field: "predoplata" = #t_predoplata; field:  "nomer_zak" = $#nci)`
![image](https://user-images.githubusercontent.com/84081305/155487744-3bbbb0ed-4f38-411d-928a-6ce0e61045e2.png)

**Reply by erpic-ru** [reply]
1. innerrowlist надо тоже передать. В data добавить:
 field: "tovari" = $innerrowlist;
2. В основную печатную форму вставить  

```
<table>
            <tr>
                <th>Наименование</th>
                <th>Кол-во</th>
                <th>Цена</th>
                <th>Сумма</th>
            </tr>
            {tovari}
        </table>
```

3. В типе шаблона печати выбрать "Элемент" (main_z) и добавить то, что передается в innerrowlist
```
                <td> {naimenovanie}
				</td>
                <td>{ordered}</td>
                <td>{cena:2|,| }</td>
                <td>{summa,number:2|,| }</td>
            </tr>

```

**Reply by Qb1cc** [reply]
Огромное спасибо, теперь я разобрался где я косячил. Всё получилось.
